.github/workflows/test.yaml

```diff
@@ -0,0 +1,343 @@
+      - name: Cache Pants Virtualenv
+        uses: actions/cache@v2
+        with:
+          path: |
+            build-support/virtualenvs
+            ~/.cache/pants/plugins/
+          key: |
+            ${{ runner.os }}-pants-venv-${{ matrix.python-version }}-${{ hashFiles('pants/3rdparty/python/**', 'pants.toml') }}
+      - name: Get Engine Hash
+        id: get-engine-hash
+        run: |
+          echo "::set-output name=hash::$(./build-support/bin/rust/print_engine_hash.sh)"
+        shell: bash
+      - name: Cache Native Engine
+        uses: actions/cache@v2
+        with:
+          path: |
+            src/python/pants/engine/internals/native_engine.so
+            src/python/pants/engine/internals/native_engine.so.metadata
+          key: |
+            ${{ runner.os }}-engine-${{ steps.get-engine-hash.outputs.hash }}
+      - name: Set env vars
+        run: |
+          echo "PANTS_CONFIG_FILES=${{ github.workspace }}/pants.travis-ci.toml" >> ${GITHUB_ENV}
+      - name: Bootstrap Pants
+        run: |
+          ./pants --version
+      - name: Upload native_engine.so
+        uses: actions/upload-artifact@v2
+        with:
+          name: native_engine.so
+          path: |
+            src/python/pants/engine/internals/native_engine.so
+            src/python/pants/engine/internals/native_engine.so.metadata
+      - name: Test Rust
+        run: |
+          # We pass --tests to skip doc tests because our generated protos contain invalid
+          # doc tests in their comments.
+          # We do not pass --all as BRFS tests don't pass on GHA MacOS containers.
+          ./cargo test --tests -- --nocapture
+        if: "!contains(env.COMMIT_MESSAGE, '[ci skip-rust]')"
+        env:
+          TMPDIR: ${{ runner.temp }}
+
+  test_python_macos:
+    name: Test Python (MacOS)
+    runs-on: macos-10.15
+    needs: bootstrap_pants_macos
+    strategy:
+      matrix:
+        python-version: [3.7.10]
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Cache Pants Virtualenv
+        uses: actions/cache@v2
+        with:
+          path: |
+            build-support/virtualenvs
+            ~/.cache/pants/plugins/
+          key: |
+            ${{ runner.os }}-pants-venv-${{ matrix.python-version }}-${{ hashFiles('pants/3rdparty/python/**', 'pants.toml') }}
+      - name: Download native_engine.so
+        uses: actions/download-artifact@v2
+        with:
+          name: native_engine.so
+          path: src/python/pants/engine/internals/
+      - name: Set env vars
+        run: |
+          echo "PANTS_CONFIG_FILES=${{ github.workspace }}/pants.travis-ci.toml" >> ${GITHUB_ENV}
+      - name: Run Python tests
+        run: |
+          ./pants test ::
```

Comment:
In Travis, we only tested "platform specific" tests (via tags). Consider doing that here to save time, although it's a tradeoff for how comprehensive we want to be.

Reply:
> In Travis, we only tested "platform specific" tests (via tags). Consider doing that here to save time, although it's a tradeoff for how comprehensive we want to be.

We only have two such tests at the moment. That doesn't justify a separate job TBH, we could just run those in the main bootstrap job.  Which is fine with me. 

I just checked on Travis and that shard does 6 minutes of setup to run 3 minutes of tests...